File: lib/api/teams.js
var _ = require('lodash'),
  Promise = require('bluebird'),
  util = require('../util'),
  log = util.logger();

function Teams(hootsuite, connection) {
  this._hootsuite = hootsuite;
  this._connection = connection;
}

Teams.prototype = {
  createTeam: function (organizationId, teamName) {
    var path = util.createPath('organizations', organizationId, 'teams');
    var data = {
      teamName: teamName
    };
    return this._connection.postJson(path, data);
  },
  appendMemberById: function (organizationId, teamId, memberId) {
    var path = util.createPath('organizations', organizationId, 'teams', teamId, 'members', memberId);
    return this._connection.post(path, { _method: 'POST' });
  },
  findByIdMembers: function (organizationId, teamId) {
    var path = util.createPath('organizations', organizationId, 'teams', teamId, 'members');
    return this._connection.get(path, { _method: 'GET' });
  },
  findMemberByIdPermissions: function (organizationId, teamId, memberId) {
    var path = util.createPath('organizations', organizationId, 'teams', teamId, 'members', memberId, 'permissions');
    return this._connection.get(path, { _method: 'GET' });
  },
  findByIdSocialProfiles: function (organizationId, teamId) {
    var path = util.createPath('organizations', organizationId, 'teams', teamId, 'socialProfiles');
    return this._connection.get(path, { _method: 'GET' });
  },
}

module.exports = Teams;

File: src.net/hootsuite-rest/Api/Teams.cs
using Newtonsoft.Json.Linq;

namespace Hootsuite.Rest.Api
{
    public class Teams
    {
        Hootsuite _hootsuite;
        Connection _connection;

        public Teams(Hootsuite hootsuite, Connection connection)
        {
            _hootsuite = hootsuite;
            _connection = connection;
        }

        public JToken createTeam(string organizationId, string teamName)
        {
            var path = util.createPath("organizations", organizationId, "teams");
            var data = new
            {
                teamName
            };
            return _connection.postJson(path, data);
        }

        public JToken appendMemberById(string organizationId, string teamId, string memberId)
        {
            var path = util.createPath("organizations", organizationId, "teams", teamId, "members", memberId);
            return _connection.post(path);
        }

        public JToken findByIdMembers(string organizationId, string teamId)
        {
            var path = util.createPath("organizations", organizationId, "teams", teamId, "members");
            return _connection.get(path);
        }

        public JToken findMemberByIdPermissions(string organizationId, string teamId, string memberId)
        {
            var path = util.createPath("organizations", organizationId, "teams", teamId, "members", memberId, "permissions");
            return _connection.get(path);
        }

        public JToken findByIdSocialProfiles(string organizationId, string teamId)
        {
            var path = util.createPath("organizations", organizationId, "teams", teamId, "socialProfiles");
            return _connection.get(path);
        }
    }
}
